src/library_wasmfs.js

```diff
@@ -143,6 +143,25 @@ FS.createPreloadedFile = FS_createPreloadedFile;
       });
     },
     // TODO: read
+    read: (fd, buffer, offset, length, position) => {
+      var seeking = typeof position != 'undefined';
+
+      var dataBuffer = _malloc(length);
+
+      var bytesRead;
+      if (seeking) {
+        bytesRead = FS.handleError(__wasmfs_pread(fd, dataBuffer, length, position));
+      } else {
+        bytesRead = FS.handleError(__wasmfs_read(fd, dataBuffer, length));
+      }
+
+      for (var i = 0; i < length; i++) {
+        buffer[i + offset] = {{{ makeGetValue('dataBuffer', 'i', 'i8')}}}
```

Comment:
```suggestion
        buffer[offset + i] = {{{ makeGetValue('dataBuffer', 'i', 'i8')}}}
```
This might seem silly, but it's a general convention to write the "fixed" part first, and then add the moving part.